Feature/lavaut m ccheat

emilelavaut opened the pull request:
Last version of the singleHit module to identify isolated blips as radiologicals for calibration.

I've add a generator tag to hit matching in order to evaluate purity and efficiency of my analysis

tomjunk commented on protoduneana/verticaldrift/singlehit/SingleHit_module.cc:
```diff
@@ -594,7 +677,7 @@ void pdvdana::SingleHit::analyze(art::Event const& e)
   int PTSIsolated = vZIsolated.size();
   std::cout << " THERE IS " << PTSIsolated << " ISOLATED POINTS IN EVENT " << fEventID << std::endl;
```
There are a lot of std::cout statements in here that look like they might be useful for debugging but will just clutter up logfiles. Is this module meant to be run in production? The message facility allows fcl steering of log levels and throttles many repeated messages, but sometimes I too prefer std::cout, if only to know that messages haven't been throttled. But ... Messages like these do not give the reader of a logfile any clue what piece of code wrote the message. If someone wants to understand what a particular message means, they have to look through all of DUNE code and LArSoft for the character string in a message. So please write the method name as part of the message, and make them optional.

tomjunk replied:
And there's a grammar error "IS ... POINTS"  (number mismatch).

tomjunk commented on protoduneana/verticaldrift/singlehit/runSingleHit_PDHD.fcl:
```diff
@@ -31,6 +31,10 @@ services:
 }
 
 services.Geometry.Name: "protodunehdv6"
+services.Geometry.GDLM: "protodunehd_v6_refactored.gdml"
```
services.Geometry.GDML, not GDLM

tomjunk commented on protoduneana/verticaldrift/singlehit/runSingleHit_PDHD.fcl:
```diff
@@ -52,18 +52,18 @@ physics:
 
-      RadiusInt:            3    # in cm
-      RadiusExt:            6    # in cm
+      RadiusInt:            0.5   # in cm
```
And a suggestion on fcl configurations -- since you have three run*.fcl files with similar parameter choices, you should make a shared fcl file with 

BEGIN_PROLOG 
  SingleHItCheaterDefaults: {
  }
END_PROLOG

and then #include that defaults file and have the line

ana: @local::SingleHItCheaterDefaults

Then you can override any default parameters in the individual run files.  This helps centralize maintenance in case you add a parameter someday, and it helps people run your module in their jobs and not just in your run* fcl jobs.

tomjunk replied:
See for example the Kalman filter track fitter default parameter fcl file for the pattern
https://github.com/LArSoft/larreco/blob/develop/larreco/TrackFinder/kalmanfilterfinaltrackfitter.fcl

aolivier23 commented on protoduneana/verticaldrift/singlehit/MCsingleHitCheater_module.cc:
```diff
@@ -0,0 +1,287 @@
+void pdvdana::MCsingleHitCheater::beginJob()
+{
+  // Implementation of optional member function here.
+  art::ServiceHandle<art::TFileService> tfs;
```
This might set off Clang when we try to cut a release.  `tfs` looks like an unused variable to me.  Can you remove it?

aolivier23 commented on protoduneana/verticaldrift/singlehit/MCsingleHitCheater_module.cc:
```diff
@@ -0,0 +1,287 @@
+  // Call appropriate consumes<>() for any products to be retrieved by this module.
+  fGeom    = &*art::ServiceHandle<geo::Geometry>();
+  auto const clockData = art::ServiceHandle<detinfo::DetectorClocksService const>()->DataForJob();
+  auto const detProp = art::ServiceHandle<detinfo::DetectorPropertiesService const>()->DataForJob(clockData);
```
This might set off Clang when we try to cut a release.  `detProp` looks like an unused variable to me.  Can you remove it without changing what the code does?

emilelavaut commented:
Thank you for your comments,
I'have put so Loglevel condition before every std::cout and changed the fcl

For the  comment on the MCCheater I also deleted the non-used objects but the module is just an intermediate module to understand the cheater not to use at production levels

tomjunk commented:
Okay, thanks for the commits and bug fixes.  I think you still have some GDLMs in there:

in protoduneana with your PR checked out:

grep -r -i gdlm *
verticaldrift/singlehit/runMCcheat_PDHD.fcl:services.Geometry.GDLM: "protodunehd_v6_refactored.gdml"
verticaldrift/singlehit/runMCcheat_PDVD.fcl:services.Geometry.GDLM: "protodunevd_v4_refactored.gdml"

emilelavaut commented:
Thank you, I fixe that now !

sam-fogarty commented:
Is it possible to merge this PR soon? @tomjunk @aolivier23 ?

aolivier23 commented:
I'll try to get it into a new release today.  It seemed just about ready the last time I looked at it.  Taking a look at the changes before I commit to that.

aolivier23 reviewed (commented):
Looks good to me.  I'm of the opinion that we should move on with doing great physics.  I'll approve as soon as I make sure it passes tests.

In the future, we could improve:
- Consider changing std::cout and LogLevel to MF_LOG_DEBUG(), mf::LogInfo(), etc. cout can be good for avoiding the automatic message squasher, but it's not clear to me that the message squasher would get in your way.

aolivier23 commented:
trigger build with DUNE/protoduneana#34

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23679&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23680&builds=)

FNALbuild commented:
:x: CI build for DUNE Failed at phase build DUNE on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/17792&builds=)

:rotating_light: For more details about the failed phase, check the [build DUNE phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/build_detail/phase_details?build_id=dune_ci/17792&platform=Linux%20slf7&phase=build&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23679&builds=)

FNALbuild commented:
:x: CI build for DUNE Failed at phase build DUNE on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/17793&builds=)

:rotating_light: For more details about the failed phase, check the [build DUNE phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/build_detail/phase_details?build_id=dune_ci/17793&platform=Linux%20slf7&phase=build&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23680&builds=)

aolivier23 commented:
Looks like I can fix the things that caused the build to fail.  Just an integer sign mismatch and a typo in a variable name.  Bear with me while I learn how to do this.  I don't think it needs any action from you.

emilelavaut commented:
thank you !  yes I made some modification on github today and I was planning to test to build it to see the errors on monday. I was not expecting you to merge today sorry.

aolivier23 commented:
@emilelavaut No problem.  Do you want me to wait until the next release to merge this pull request?  I'm going to cut a release today to match the new LArSoft version.  Sounds like the next release will be next Friday.

sam-fogarty commented:
@aolivier23 @emilelavaut I may suggest waiting until next Friday as there is at least one change we are going to add (getting absolute timestamps for the clusters) which is relevant for charge -light matching. So we might as well wait, and maybe Emile will have a few more small changes he may want to add.

emilelavaut commented:
Yes next friday would be perfect for me !

emilelavaut commented:
this version of the code work well and have everything updated.
it has been test and it should be able to merge.

aolivier23 commented:
Thanks @emilelavaut .  I'm going to try to merge this PR in tomorrow afternoon.  Speak now if you want me to hold off please!

aolivier23 commented:
trigger build with dune/protoduneana#34

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23736&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23737&builds=)

FNALbuild commented:
:x: CI build for DUNE Failed at phase build DUNE on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/17847&builds=)

:rotating_light: For more details about the failed phase, check the [build DUNE phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/build_detail/phase_details?build_id=dune_ci/17847&platform=Linux%20slf7&phase=build&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23737&builds=)

FNALbuild commented:
:x: CI build for DUNE Failed at phase build DUNE on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/17846&builds=)

:rotating_light: For more details about the failed phase, check the [build DUNE phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/build_detail/phase_details?build_id=dune_ci/17846&platform=Linux%20slf7&phase=build&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23736&builds=)

aolivier23 reviewed (approved):
Looks good!  A few FHICL files changed to resolve Tom's comment about using an include file.  Very readable now.

Note that I resolved a merge conflict in the module.  I updated a function name to match the new LArSoft version's breaking change.
